src/editors/containers/TextEditor/components/ImageSettingsModal/hooks.js

```diff
@@ -0,0 +1,251 @@
+ * dimensions.
+ * @param {obj} dimensions - current stored dimensions
+ * @param {obj} local - local (active) dimensions in the inputs
+ * @param {obj} locked - locked dimensions
+ * @return {obj} - output dimensions after attempted move ({ height, width })
+ */
+export const newDimensions = ({ dimensions, local, locked }) => (
+  (!locked || checkEqual(local, dimensions))
+    ? local
+    : module.getValidDimensions({ dimensions, local, locked })
+);
+
+/**
+ * lockDimensions({ dimensions, lockInitialized, setLocked })
+ * Lock dimensions if lock initialized.  Store minimum valid increment on lock so
+ * that we don't have re-compute.
+ * @param {obj} dimensions - current stored dimensions
+ * @param {bool} lockInitialized - has the lock state initialized?
+ * @param {func} setLocked - set lock state
+ */
+export const lockDimensions = ({ dimensions, lockInitialized, setLocked }) => {
+  if (!lockInitialized) { return; }
+
+  // find minimum viable increment
+  let gcd = findGcd(dimensions.width, dimensions.height);
+  if ([dimensions.width, dimensions.height].some(v => !Number.isInteger(v / gcd))) {
+    gcd = 1;
+  }
+  const minInc = { width: dimensions.width / gcd, height: dimensions.height / gcd, gcd };
+  setLocked({ ...dimensions, minInc });
+};
+
+/**
+ * dimensionLockHooks({ dimensions })
+ * Returns a set of hooks pertaining to the dimension locks.
+ * Locks the dimensions initially, on lock initialization.
+ * @param {obj} dimensions - current stored dimensions
+ * @return {obj} - dimension lock hooks
+ *   {func} initializeLock - enable the lock mechanism
+ *   {obj} locked - current locked state
+ *   {func} lock - lock the current dimensions
+ *   {func} unlock - unlock the dimensions
+ */
+export const dimensionLockHooks = ({ dimensions }) => {
+  const [locked, setLocked] = module.state.locked(null);
+  const [lockInitialized, setLockInitialized] = module.state.lockInitialized(null);
+  const lock = () => module.lockDimensions({ lockInitialized, dimensions, setLocked });
+
+  React.useEffect(lock, [lockInitialized]);
+
+  return {
+    initializeLock: () => setLockInitialized(true),
+    locked,
+    lock,
+    unlock: () => setLocked(null),
+  };
+};
+
+/**
+ * dimensionHooks()
+ * Returns an object of dimension-focused react hooks.
+ * @return {obj} - dimension hooks
+ *   {func} onImgLoad - initializes image dimension fields
+ *     @param {object} selection - selected image object with possible override dimensions.
+ *     @return {callback} - image load event callback that loads dimensions.
+ *   {object} locked - current locked state
+ *   {func} lock - lock current dimensions
+ *   {func} unlock - unlock dimensions
+ *   {object} value - current dimension values
+ *   {func} setHeight - set height
+ *     @param {string} - new height string
+ *   {func} setWidth - set width
+ *     @param {string} - new width string
+ *   {func} updateDimensions - set dimensions based on state
+ */
+export const dimensionHooks = () => {
```

Comment:
This will have to initialize with either null, or some value (from image selected from editor, as we need to respect the html). Shouldn't be too much of a change, but wanted to point out this  (+ same for alt text) is where our two PRs collide.